pgjdbc/src/main/java/org/postgresql/util/PSQLState.java

```diff
@@ -118,12 +121,29 @@ public String getState() {
     return this.state;
   }
 
-  public static boolean isConnectionError(@Nullable String psqlState) {
-    return PSQLState.CONNECTION_UNABLE_TO_CONNECT.getState().equals(psqlState)
-        || PSQLState.CONNECTION_DOES_NOT_EXIST.getState().equals(psqlState)
-        || PSQLState.CONNECTION_REJECTED.getState().equals(psqlState)
-        || PSQLState.CONNECTION_FAILURE.getState().equals(psqlState)
-        || PSQLState.CONNECTION_FAILURE_DURING_TRANSACTION.getState().equals(psqlState);
+  private static Map<String, PSQLState> codeToEnum;
+
+  @Nullable
+  public static PSQLState fromCode(final String string) {
+    if (codeToEnum == null) {
```

Comment:
I think the worst case would be multiple constructions of identical `Map`s since the field `codeToEnum` will be either `null` or set to the completely filled map `temp`, but I could also use synchronization instead. Or maybe I could initialize the map in a `static {}` block instead.

Reply:
I believe that would only be true if codeToEnum were final or volatile. Since it is neither, it is perfectly valid for a partially constructed object to be observed from another thread.

Reply:
Yes, initialize it in a static block, or in the constructor.